### PR TITLE
AP_OSD: add calendar date panel

### PR DESCRIPTION
**Pull request by vierfuffzig**
this adds a date panel in response to https://github.com/ArduPilot/ardupilot/issues/12706
auto switch functionality might optionally be added, for now a standard type panel offers most flexibility for testing imho.

### CODE REVIEW

File: libraries/AP_RTC/AP_RTC.cpp [hunk 1]
```diff
@@ -125,12 +125,50 @@ bool AP_RTC::get_local_time(uint8_t &hour, uint8_t &min, uint8_t &sec, uint16_t
...
+    
 
     // convert times as milliseconds into appropriate units
     sec = sec_ms / 1000;
     min = min_ms / (60 * 1000);
     hour = hour_ms / (60 * 60 * 1000);
-
+    
+    // calculate human calendar date
+    const uint32_t secs_day   =  86400; //  24* 60 * 60
```
Comment (tridge) [inline libraries/AP_RTC/AP_RTC.cpp]: we actually already have a function to parse dates here:
https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/hwdef/common/stm32_util.c#L144
I think we should move that function to AP_RTC, and then use the common function in both places. 
Is that something you are happy to do or should I do it?

Comment (vierfuffzig) [reply libraries/AP_RTC/AP_RTC.cpp]: @tridge uh. i grew some grey hair trying to get that date-math right... is it ok if i try to move that existing function and cry for your help when i get stuck?

**Comment by khancyr** [comment]
@vierfuffzig hello, do you want to pursuit this ?

**Comment by vierfuffzig** [comment]
@khancyr i don't think so. when thinking back i can't really remember i ever missed a calender date panel. on a sidenote, moving the date function from HAL to RTC as @tridge suggested is way beyond my pacheck range. thanks for getting back on this, but i'll close this.

**Pull request closed without merging**